galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/common/PropertyUtilities.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.framework.api.cps.internal.common;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.validation.constraints.NotNull;
+
+import com.google.gson.Gson;
+
+import dev.galasa.framework.api.common.InternalServletException;
+import dev.galasa.framework.api.common.ServletError;
+import dev.galasa.framework.spi.ConfigurationPropertyStoreException;
+import dev.galasa.framework.spi.FrameworkException;
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.utils.GalasaGsonBuilder;
+
+import static dev.galasa.framework.api.common.ServletErrorMessage.*;
+
+public class PropertyUtilities {
+
+    private static final Set<String> updateActions = Set.of("apply","update");
+
+    static final Gson gson = GalasaGsonBuilder.build();
+    
+    IFramework framework;
+
+    private static final String REDACTED_PROPERTY_VALUE = "********";
+
+    private static final Set<String> hiddenNamespaces = new HashSet<>();
+    static {
+        hiddenNamespaces.add("dss");
+    }
+
+    /**
+     * Some namespaces are able to be set, but cannot be queried.
+     *
+     * When they are queried, the values are redacted
+     */
+    private static final Set<String> secureNamespaces = new HashSet<>();
+    static {
+        secureNamespaces.add("secure");
+    }
+
+    public PropertyUtilities(IFramework framework){
+        this.framework = framework;
+    }
+
+    private IFramework getFramework(){
+        return this.framework;
+    }
+
+
+    public boolean isPropertyValid(GalasaProperty property) throws InternalServletException {
```

Comment:
We already have an `isPropertyValid()` method in `GalasaProperty` so we probably don't need this here

Reply:
Shouldn't the isPropertyValid be a method on the GalasaProperty ? rather than being some utility class method ?